code/datums/flock/flock.dm

```diff
@@ -154,18 +156,19 @@
 /datum/flock/proc/total_health_percentage()
 	var/hp = 0
 	var/max_hp = 0
-	for(var/mob/living/critter/flock/F as anything in src.units)
-		F.count_healths()
-		hp += F.health
-		max_hp += F.max_health
+	for(var/pathkey in src.units)
+		for(var/mob/living/critter/flock/F as anything in src.units[pathkey])
+			F.count_healths()
+			hp += F.health
+			max_hp += F.max_health
 	if(max_hp != 0)
 		return hp/max_hp
 	else
 		return 0
 
 /datum/flock/proc/total_resources()
 	. = 0
-	for(var/mob/living/critter/flock/F as anything in src.units)
+	for(var/mob/living/critter/flock/F as anything in src.units[/mob/living/critter/flock/drone])
```

Comment:
```suggestion
	for(var/mob/living/critter/flock/drone/F as anything in src.units[/mob/living/critter/flock/drone])
```
Not necessary, but maybe since only expecting drones here?

Reply:
Yeah alright, but it'll need undoing if we ever add another flock critter that collects resources and isn't a drone